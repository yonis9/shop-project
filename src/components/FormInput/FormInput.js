import React from 'react';

import { 
    GroupContainer,
    FormInputContainer,
    FormLabel,
} from './FormInput.styles';

const FormInput = ({ handleChange, label, ...otherProps }) => (
    <GroupContainer>
        <FormInputContainer onChange={handleChange} {...otherProps} />
        {
            label ?
            (<FormLabel className={otherProps.value.length ? 'shrink' : ''}>
                {label}
            </FormLabel>) : null
        }
    </GroupContainer>
) 

export default FormInput;