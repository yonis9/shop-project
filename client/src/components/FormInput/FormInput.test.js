import React from 'react';
import { shallow } from 'enzyme';

import FormInput from './FormInput';


describe('FormInput component', () => {
    let wrapper;
    let mockHandleChange;

    beforeEach(() => {
        mockHandleChange = jest.fn();

        const mockProps = {
            label: 'email',
            value: 'jhon@gmail.com',
            handleChange: mockHandleChange,
        };

        wrapper = shallow(<FormInput {...mockProps}/>);
    });

    it('should rendered FormInput component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should call handleChange when input changes', () => {
        wrapper.find('FormInputContainer').simulate('change');

        expect(mockHandleChange).toHaveBeenCalled();
    });

    it('should render FormLabel if there is a label', () => {
        expect(wrapper.exists('FormLabel')).toBe(true);
    });

    it('should not render FormLabel if there is no label', () => {
        const newMockProps = {
            label: '',
            value: 'jhon@gmail.com',
            handleChange: mockHandleChange,
        };

        const newWrapper = shallow(<FormInput {...newMockProps}/>);
        
        expect(newWrapper.exists('FormLabel')).toBe(false);
    });
    
});