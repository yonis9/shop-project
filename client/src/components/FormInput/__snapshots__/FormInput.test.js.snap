// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput component should rendered FormInput component 1`] = `ShallowWrapper {}`;
